### PR TITLE
Move Dev Home to WAP Project to support upcoming features

### PR DESCRIPTION
**Pull request by jaholme**
## Summary of the pull request
This change moves the Dev Home project over to WAP Project to support upcoming features.
The main changes are the way the build.ps1 script generates the msix packages as well as a couple of unfortunate changes needed to the AppXManifest to explicitly list resource languages supported.

## Validation steps performed
Ran build pipeline and validated that the installed package was correct.
Verified sources are localized correctly.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: Packaging/Package.appxmanifest [hunk 1]
```diff
@@ -81,23 +95,11 @@
           </com:ComServer>
         </com:Extension>
         <uap3:Extension Category="windows.appExtension">
-          <uap3:AppExtension Name="com.microsoft.devhome" Id="PG-SP-ID1" PublicFolder="Public" DisplayName="ms-resource:AppDisplayNameCoreExt" Description="ms-resource:AppDescriptionCoreExt">
+          <uap3:AppExtension Name="com.microsoft.devhome" Id="PG-SP-ID1" PublicFolder="Public" DisplayName="ms-resource:AppDisplayNameDev" Description="ms-resource:AppDescription">
```
Comment (krschau) [inline Packaging/Package.appxmanifest:98]: I think there may have been some merge conflicts in this file from here down, widget stuff should go back.

Comment (jaholme) [reply Packaging/Package.appxmanifest:98]: ah thanks for catching this, the build.ps1 changes some of this every time its run and doesn't put it back.

File: Packaging/Packaging.wapproj [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup>
+    <WapProjPath Condition="'$(WapProjPath)'==''">$(MSBuildExtensionsPath)\Microsoft\DesktopBridge\</WapProjPath>
+  </PropertyGroup>
+  <Import Project="$(WapProjPath)\Microsoft.DesktopBridge.props" />
+  <PropertyGroup>
+    <ProjectGuid>05d4c6e5-d33d-4c15-91fe-3dd2b6a5d61c</ProjectGuid>
+    <TargetPlatformVersion>10.0.22621.0</TargetPlatformVersion>
+    <TargetPlatformMinVersion>10.0.19041.0</TargetPlatformMinVersion>
```
Comment (krschau) [inline Packaging/Packaging.wapproj:56]: ToolingVersions.props is imported at the top so we don't have to set this here.

File: HyperVExtension/src/HyperVExtension/HyperVExtension.csproj [hunk 3]
```diff
@@ -54,6 +54,10 @@
     <PackageReference Include="Microsoft.Extensions.Http" Version="8.0.0" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (krschau) [inline HyperVExtension/src/HyperVExtension/HyperVExtension.csproj:57]: If all resw files need to be included this way, please add a step in https://github.com/microsoft/devhome/blob/main/docs/tools/creating-a-tool.md after the step explaining how to create the resw file.

File: HyperVExtension/BuildDevSetupAgentHelper.ps1 [hunk 4]
```diff
@@ -80,6 +80,7 @@ Try {
 
   $msbuildArgs = @(
       ("HyperVExtension\DevSetupAgent.sln"),
+      ("/m"),
```
Comment (dhoehna) [inline HyperVExtension/BuildDevSetupAgentHelper.ps1:83]: I've always ran into problems building with `/m` on the official builds.  But, if it works here.  Cool.

**Review by dhoehna — APPROVED** [review]

File: HyperVExtension/src/HyperVExtension/HyperVExtension.csproj (hunk 3 above)
Comment (dhoehna) [reply HyperVExtension/src/HyperVExtension/HyperVExtension.csproj:57]: Can this be in a props file?

**Review by zadesai — APPROVED** [review]

File: HyperVExtension/src/HyperVExtension/HyperVExtension.csproj (hunk 3 above)
Comment (jaholme) [reply HyperVExtension/src/HyperVExtension/HyperVExtension.csproj]: Looks like this was unnecessary. I'm re-running the pipeline to be sure but thanks for pointing this out.

File: HyperVExtension/BuildDevSetupAgentHelper.ps1 (hunk 4 above)
Comment (jaholme) [reply HyperVExtension/BuildDevSetupAgentHelper.ps1:83]: Seems to work in official builds, its a shame its not compatible with WAP Project

File: Packaging/Packaging.wapproj (hunk 2 above)
Comment (jaholme) [reply Packaging/Packaging.wapproj:56]: I'll add a comment, but unfortunately toolingversions.props doesn't work in wapproj,

 F:\DevHomeInternal\Packaging\Packaging.wapproj : error NU1201: Project DevHome is not compatible with uap10.0.19041 (
UAP,Version=v10.0.19041) / win10-x86-aot. Project DevHome supports: net8.0-windows10.0.22621 (.NETCoreApp,Version=v8.0)
 [F:\DevHomeInternal\DevHome.sln]

**Review by krschau — APPROVED** [review]

**Review by EricJohnson327 — APPROVED** [review]

**Pull request merged**